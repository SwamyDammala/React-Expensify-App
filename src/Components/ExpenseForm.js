import React from 'react'
import moment from 'moment'
import { SingleDatePicker} from 'react-dates'


const now=moment();
// console.log(now.format('MMM Do, YYYY'));


class ExpenseForm extends React.Component{
    constructor(props){
        super(props)    
            this.state={
                    description:props.expense?props.expense.description:'',
                    notes:props.expense?props.expense.notes:'',
                    amount:props.expense?(props.expense.amount/100).toString():'',
                    createdAt:props.expense?moment(props.expense.createdAt):moment(),
                    calendarFocused:false,
                    error:''
                }
        }
     
    handleDescription=(e)=>{
        const description=e.target.value
        this.setState(()=>({description}))
    }

    handleChange=(e)=>{
        const amount=e.target.value
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
            this.setState(()=>({amount}))
        }        
    }

    onDateChange=(createdAt)=>{
        if(createdAt){
            this.setState(()=>({createdAt}))
        }
    }

    onFocusChange=({focused})=>{
        this.setState(()=>({calendarFocused:focused}))
    }

    handleNotes=(e)=>{
        const notes=e.target.value
        this.setState(()=>({notes}))
    }

    handleSubmit=(e)=>{
        e.preventDefault();
        if(!this.state.description || !this.state.amount){
            this.setState(()=>({error:'Please provide the description and amount'}))
        }
        else{
            this.setState(()=>({error:''}))
            this.props.onSubmit({
                description:this.state.description,
                amount:parseFloat(this.state.amount,10)*100,
                createdAt:this.state.createdAt.valueOf(),
                notes:this.state.notes
            })
        }
    }

    render(){
        return(
            <div>
                
                <form className='form' onSubmit={this.handleSubmit}>
                    {this.state.error && <p className='form__error'>{this.state.error}</p>}
                    <input type='text' placeholder='Description'
                     className='text-input'
                     autoFocus value={this.state.description} 
                     onChange={this.handleDescription}/>
                    <input type='number' placeholder='Amount' 
                     className='text-input'
                    value={this.state.amount} onChange={this.handleChange}/>
                    <SingleDatePicker 
                    date={this.state.createdAt}
                    onDateChange={this.onDateChange}
                    focused={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}//show the previous month dates...
                    isOutsideRange={(()=>false)} //it will
                    />
                     <textarea placeholder='Add a note for your expense'
                      className='textarea' 
                     value={this.state.notes} onChange={this.handleNotes}></textarea>
                     <div>
                     <button className='button'>Save Expense</button>
                     </div>
                </form>
            </div>
        )
    }
}

export default ExpenseForm




