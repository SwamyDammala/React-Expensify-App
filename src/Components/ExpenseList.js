import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import SelectExpenses from '../selectors/expenses'

export const ExpenseList=(props)=>(
    <div className='content-container'>
        <div className='list-header'>
            <div className='show-for-mobile'>Expenses</div>
            <div className='show-for-desktop'>Expense</div>
            <div className='show-for-desktop'>Amount</div>
        </div>
        <div className='list-body'>
        {
            props.expenses.length===0?
            <div className='list-item list-item--messege'>
                <span>No Expenses</span>
            </div>:
            props.expenses.map(expense=><ExpenseListItem key={expense.id} {...expense} />)
        }
        </div>
        
    </div>
)

// const ConnectedExpensesList=connect((state)=>{
//     return {
//         name:'swamy'
//     }
// })(ExpanseList)

const MapStateToProps=((state)=>{
    return {
        expenses:SelectExpenses(state.expenses,state.filters)
    }
})

export default connect(MapStateToProps)(ExpenseList)

