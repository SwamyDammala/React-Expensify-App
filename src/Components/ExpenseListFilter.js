import React from 'react'
import { connect } from 'react-redux'
import { setTextFilter , setSortByDate , setSortByAmount , setStartDate , setEndDate} from '../actions/filters'
import { DateRangePicker } from 'react-dates'


export class ExpenseListFilter extends React.Component{
    constructor(props){
        super(props)
        this.state={
            calendarFocused:null
        }
    }

    onDatesChange=({startDate,endDate})=>{
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange=(calendarFocused)=>{
        this.setState(()=>({calendarFocused}))
    }
    onTextChange=(e)=>{
        this.props.setTextFilter(e.target.value)
                }
    onSortByChange=(e)=>{
        if(e.target.value==='date'){
            this.props.setSortByDate()
        }
        else if(e.target.value==='amount'){
            this.props.setSortByAmount()
        }
    }

    render(){
        return(
            <div>
                <input type='text' value={this.props.filters.text} onChange={this.onTextChange}/>
        
                <select value={this.props.filters.sortBy} onChange={this.onSortByChange}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker 
                startDate={this.props.filters.startDate}
                endDate={this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calendarFocused} 
                onFocusChange={this.onFocusChange} 
                showClearDates={true} //It will show the clear button at end
                numberOfMonths={1}
                isOutsideRange={()=>{false}}
                //initialVisibleMonth={() => moment().add(2, "M")}
                />
            </div>
        )          
    }
}


const MapStateToProps=((state)=>{
    return{
        filters:state.filters
        }
})

const mapsDispatchToProps=(dispatch)=>{
    return{
        setTextFilter:(text)=>dispatch(setTextFilter(text)),
        setSortByAmount:()=>dispatch(setSortByAmount()),
        setSortByDate:()=>dispatch(setSortByDate()),
        setStartDate:(startDate)=>dispatch(setStartDate(startDate)),
        setEndDate:(endDate)=>dispatch(setEndDate(endDate))
    }
}


export default connect(MapStateToProps,mapsDispatchToProps)(ExpenseListFilter)