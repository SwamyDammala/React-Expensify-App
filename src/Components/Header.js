import React from 'react'
import { NavLink } from 'react-router-dom'

const Header=()=>(
    <header>
        <div>
            <h1>Expensify</h1>  
            <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink><br/>
            <NavLink to="/create" activeClassName="is-active">Create expense</NavLink><br/>
            <NavLink to="/edit" activeClassName="is-active">Edit expense Page</NavLink><br/>
            {/* <NavLink to="/help" activeClassName="is-active">Help</NavLink><br/> */}
        </div>
    </header>
);

export default Header