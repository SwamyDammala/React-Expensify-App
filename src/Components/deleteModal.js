import React from 'react'
import Modal from 'react-modal'

const DeleteModal=(props)=>(
    <Modal 
            isOpen={props.deleteConfim}
            onRequestClose={props.handleModal}
            contentLabel='Delete Expense'
            className='modal'>
        
            <p className='modal__body'>Do you want to remove this item from expense tracker?</p>
            <button onClick={props.handleModal}>Cancel</button>
            <button onClick={props.remove}>Delete</button>
    </Modal>
)

export default DeleteModal

