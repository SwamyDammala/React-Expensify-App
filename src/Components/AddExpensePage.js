import React from 'react'
import ExpenseForm from './ExpenseForm'
import { connect } from 'react-redux'
import { startAddExpense } from '../actions/expenses'


export class AddexpensePage extends React.Component{
    onSubmit=(expense)=>{
        this.props.startAddExpense((expense))
        this.props.history.push('/dashboard')
    }
    render(){
        return(
            <div>
            <h3>This is new expense page</h3>
            <ExpenseForm onSubmit={this.onSubmit} />
            </div>
        )
    }
}

//This is firstway to dispatch the add expense method with out using maps to props
// const AddexpensePage=(props)=>(
//     <div>
//         <h3>This is new expense page</h3>
//         <ExpenseForm onSubmit={(expense)=>{
//             props.dispatch(addexpense(expense))
//             props.history.push('/')
//         }} />
//     </div>    
// )

const mapsDispatchToProps=(dispatch)=>({
    startAddExpense:(expense)=>dispatch(startAddExpense(expense))
})

export default connect(undefined,mapsDispatchToProps)(AddexpensePage)