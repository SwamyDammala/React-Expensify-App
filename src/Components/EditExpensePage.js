import React from 'react'
import ExpenseForm from './ExpenseForm'
import { startEditExpenses } from '../actions/expenses'
import { connect } from 'react-redux'
import { startRemoveExpenses } from '../actions/expenses'
import expenses from '../tests/fixtures/expenses'

export class EditexpensePage extends React.Component{
    onSubmit=(expense)=>{
        this.props.startEditExpenses(this.props.expense.id,expense)
        this.props.history.push('/dashboard')
    }
    onRemove=()=>{
        this.props.startRemoveExpenses({id:this.props.expense.id})
        this.props.history.push('/dashboard')
    }
    render(){
        return(
            <div>
            <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit}/>
                <button onClick={this.onRemove}>Remove Expense</button>
                </div>
        )
    }
}

// const EditexpensePage=(props)=>{
//     return(
//         <div>            
//             <ExpenseForm expense={props.expenses}
//                     onSubmit={(expense)=>{
//                         props.dispatch(editexpense(props.expenses.id,expense))
//                         console.log('updated',expense)
//                         props.history.push('/')
//             }}/>
//                 <button onClick={()=>{props.dispatch(removeexpense({id:props.expenses.id}))
//                     props.history.push('/')
//                     }}>Remove Expense</button>
//      </div>
//     ) 
// }

const MapsStoreToProps=(store,props)=>{
    return{
        expense:store.expenses.find((expense)=>(expense.id===props.match.params.id))
    }
}

const mapsDispatchToProps=(dispatch,props)=>{
    return{ 
        startEditExpenses:(id,expense)=>dispatch(startEditExpenses(id,expense)),
        startRemoveExpenses:(id)=>dispatch(startRemoveExpenses(id))
    }

}

export default connect(MapsStoreToProps,mapsDispatchToProps)(EditexpensePage)
