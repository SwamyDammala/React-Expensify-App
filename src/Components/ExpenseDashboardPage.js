import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilter from './ExpenseListFilter'
import ExpenseSummary from './ExpenseSummary'
import DeleteModal from './deleteModal'

const ExpenseDashboardPage=()=>(
    <div>
        <ExpenseSummary />
        <ExpenseListFilter />
        <ExpenseList />
    </div>    
)

export default ExpenseDashboardPage