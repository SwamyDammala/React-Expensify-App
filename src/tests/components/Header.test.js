import React from 'react'
import {shallow} from 'enzyme'
import {Header} from '../../Components/Header'


test('Chechk header component using Shallow snapshot method',()=>{
    const wrapper=shallow(<Header startLogout={{}={}}/>)
    expect(wrapper).toMatchSnapshot()
    // const renderer=new ReactShallowRenderer()
    // renderer.render(<Header />)
    // expect(renderer.getRenderOutput()).toMatchSnapshot()
})

test('check header component using actual logout button',()=>{
    const startLogout=jest.fn()
    const wrapper=shallow(<Header startLogout={startLogout}/>)
    wrapper.find('button').simulate('click')
    expect(startLogout).toHaveBeenCalled()
})