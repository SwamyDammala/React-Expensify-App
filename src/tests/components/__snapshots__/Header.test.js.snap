// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chechk header component using Shallow snapshot method 1`] = `
<header>
  <div>
    <h1>
      Expensify
    </h1>
    <NavLink
      activeClassName="is-active"
      exact={true}
      to="/"
    >
      Dashboard
    </NavLink>
    <br />
    <NavLink
      activeClassName="is-active"
      to="/create"
    >
      Create expense
    </NavLink>
    <br />
  </div>
</header>
`;
